File: common/src/main/java/com/ssblur/scriptor/word/descriptor/SpeedDurationDescriptor.java
package com.ssblur.scriptor.word.descriptor;

import com.ssblur.scriptor.api.word.descriptor.SpeedDescriptor;
import com.ssblur.scriptor.word.descriptor.duration.SimpleDurationDescriptor;

public class SpeedDurationDescriptor extends SimpleDurationDescriptor implements SpeedDescriptor {
  double speed;

  public SpeedDurationDescriptor(int cost, double duration, double speed) {
    super(cost, duration);
    this.speed = speed;
  }

  @Override
  public double speedModifier() {
    return speed;
  }
}

File: common/src/main/kotlin/com/ssblur/scriptor/word/descriptor/SpeedDurationDescriptor.kt
package com.ssblur.scriptor.word.descriptor

import com.ssblur.scriptor.word.descriptor.duration.SimpleDurationDescriptor

class SpeedDurationDescriptor(cost: Int, duration: Double, var speed: Double) :
    SimpleDurationDescriptor(cost, duration), SpeedDescriptor {
    override fun speedModifier(): Double {
        return speed
    }
}
